files/nodebot.rb

```diff
@@ -57,5 +60,101 @@ def handle
   def send(channel, message)
     system('nodebot', channel, message)
   end
+
+end
+
+ANSI_TO_IRC_COLORS = {
+  30 => "01",  # black
+  31 => "04",  # red
+  32 => "09",  # green
+  33 => "08",  # yellow
+  34 => "02",  # blue
+  35 => "13", # pink
+  36 => "11", # cyan
+  37 => "00",  # white
+  39 => "00",   # white
+}
+# Background colors
+ANSI_TO_IRC_BACK = {}
+# Formatting characters
+ANSI_TO_IRC_FMT = {
+  0  => 0xf,  # reset
+  1  => 0x2,  # bold
+  4  => 0x1f, # underline
+  7  => 0x16, # reverse mode
+}
+# Generate background colors
+ANSI_TO_IRC_COLORS.each { |k, v|
+  ANSI_TO_IRC_BACK[k + 10] = v
+}
+ANSI_START = Regexp.new(Regexp.quote("\x1b"))
+
+def ansi_to_irc_colors(message)
+  output = ""
+  scanner = StringScanner.new(message)
+
+  token = scanner.scan_until(ANSI_START)
+  # No ANSI sequences :(
+  if !token
+     return message
+  end
+
+  while token
+    # Remove the ANSI escape sequence
+    # and append whatever is left to
+    # the output
+    output += token[0...-1]
+
+    next_byte = scanner.get_byte()
+    if next_byte >= "0x40" and next_byte <= "0x7E"
+      # Already reached the end of the sequence?
+      # Do nothing.
+    elsif next_byte == "["
+      sequence = scanner.scan_until(/[\x40-\x7E]/)
+      # Discard the final byte
+      sequence = sequence[0...-1]
+      # Split by semicolon
+      sequence = sequence.split(";")
+
+      foreground = nil
+      background = nil
+      formatting = nil
+      sequence.each { |ansi_code|
+          ansi_code = Integer(ansi_code) rescue nil
+          if ansi_code
+            foreground ||= ANSI_TO_IRC_COLORS[ansi_code]
+            background ||= ANSI_TO_IRC_BACK[ansi_code]
+            formatting ||= ANSI_TO_IRC_FMT[ansi_code]
+          end
+      }
+
+      if !foreground.nil?
```

Comment:
here and in other places: `if !foo.nil?` is identical to just `if foo` unless you want this condition to eval to `true` for `false` value of `foo`, which you don't in this case